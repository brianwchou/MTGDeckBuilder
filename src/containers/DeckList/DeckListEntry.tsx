import React from 'react';

export default function DeckListEntry ({index, card, count, getCardInfo}) {
    const handleOnClick = (e) => {
        getCardInfo(card, e.target.name);
    }

    return (
        <div className="flex alignment" key={index}> 
            <div className="flex count">
                {count}x 
                <button onClick={handleOnClick} name={'increment'}>+</button>
            </div>
            <div className="flex name">
                {card.name} 
                <button onClick={handleOnClick} name={'decrement'}>-</button>
                <button onClick={handleOnClick} name={'maybe'}>maybe</button>
            </div>        
        </div>
    )
}