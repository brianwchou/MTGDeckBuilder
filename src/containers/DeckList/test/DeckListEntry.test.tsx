import React from 'react'
import DeckListEntry from '../../DeckList/DeckListEntry';
import renderer from 'react-test-renderer'
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('DeckListEntry', () => {
    
    test('rendering correctly', () => {
        const mockNoCard = {}
        const tree = renderer
        .create(<DeckListEntry card={mockNoCard} />)
        .toJSON();

    expect(tree).toMatchSnapshot();
    });

    test('clicking a button calls getCardInfo', () => {
        const mockCard = {name: "thing in the ice"}
        const mockTarget = {target: { name: "increment"} }
        const onClickSpy = jest.fn()
        const wrapper = shallow(<DeckListEntry card={mockCard} getCardInfo={onClickSpy} />)

        let container = wrapper.find("div")
        let containerButton = container.find("button").at(0)

        expect(onClickSpy).not.toHaveBeenCalled();
        containerButton.simulate('click', mockTarget);
        expect(onClickSpy).toHaveBeenCalled();
    });
});