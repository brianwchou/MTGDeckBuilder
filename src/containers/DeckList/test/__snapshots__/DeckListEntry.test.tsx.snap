// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeckListEntry rendering correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container"
>
  <div>
    x
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
      disabled={false}
      name="increment"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        +
      </span>
    </button>
  </div>
  <div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
      disabled={false}
      name="decrement"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        -
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
      disabled={false}
      name="maybe"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        maybe
      </span>
    </button>
  </div>
</div>
`;
